openff/toolkit/topology/molecule.py

```diff
@@ -4595,31 +4660,73 @@ def canonical_order_atoms(self, toolkit_registry=GLOBAL_TOOLKIT_REGISTRY):
                 f"Got {type(toolkit_registry)}."
             )
 
-    def remap(self, mapping_dict, current_to_new=True):
+    def remap(
+        self,
+        mapping_dict: Dict[int, int],
+        current_to_new: bool = True,
+        partial: bool = False,
+    ):
         """
-        Remap all of the indexes in the molecule to match the given mapping dict
+        Reorder the atoms in the molecule according to the given mapping dict.
+
+        The mapping dict must be a dictionary mapping atom indices to atom
+        indices. Each atom index must be an integer in the half-open interval
+        ``[0, n_atoms)``. All positions in the molecule must be mapped from and
```

Comment:
(not blocking) I think this is clearer than saying "positions", since that could be confused with coordinates.

It may be good if you you also define "atom index" somewhere. It's fine if this needs to be repeated in multiple docstrings.

```suggestion
        ``[0, n_atoms)``. All atom indices in the molecule must be mapped from and
```